03week/towersOfHanoi.js

```diff
@@ -2,49 +2,90 @@
 
 const assert = require('assert');
 const readline = require('readline');
-const rl = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout
-});
+const rl = readline.createInterface({input: process.stdin, output: process.stdout});
+let removedNumber = 0;
+let pushedNumber = 0;
 
 let stacks = {
-  a: [4, 3, 2, 1],
+  a: [
+    4, 3, 2, 1
+  ],
   b: [],
   c: []
 };
 
+function resetGame(reset) {
+  if (reset === 'yes') {
+    stacks = {
+      a: [
+        4, 3, 2, 1
+      ],
+      b: [],
+      c: []
+    };
+    getPrompt()
+  } else {
+    console.log('ctrl + c to quit')
+
+  }
+}
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack, endStack) {
   // Your code here
-
+  if (stacks[startStack] && stacks[endStack]) {
+    removedNumber = stacks[(startStack)].pop();
+    pushedNumber = stacks[(endStack)].push(removedNumber) - 2
+    isLegal(startStack, endStack);
+
+  } else {
+    getPrompt()
+  }
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(startStack, endStack) {
+  if (removedNumber > stacks[(endStack)][pushedNumber]) {
+    stacks[(startStack)].push(stacks[(endStack)].pop());
+    getPrompt();
+  }
+  if (!removedNumber) {
+    stacks[(endStack)].pop();
+    getPrompt();
+  } else {
+    checkForWin();
+  }
 }
 
 function checkForWin() {
   // Your code here
+  if (stacks.b.length === 4 || stacks.c.length === 4) {
+    console.log('!!!!!!   You Win  !!!!!!');
+    rl.question("Would you like to reset?", (reset) => {
+      resetGame(reset);
+    });
+
+  } else {
+    getPrompt();
+  }
 
 }
 
 function towersOfHanoi(startStack, endStack) {
   // Your code here
```

Comment:
your towersOfHanoi function should call all of your functions in order of necessity. For example, if isLegal() is true then you would call movePieceFunction() then you would check for a win.